File: Shopflow/shopflow_front/src/Modeles/Coupon.ts
// ============================================================
// Modèles Coupon — synchronisés avec le backend ShopFlow
// CouponController : /api/coupons
// ============================================================

import { CouponType } from './enums';

// ─── Request DTO ─────────────────────────────────────────────

export interface CouponRequest {
  code: string;
  type: CouponType;
  value: number;
  expirationDate: string;   // ISO 8601
  maxUsages: number;
}

// ─── Response DTO ────────────────────────────────────────────

export interface CouponResponse {
  id: number;
  code: string;
  type: CouponType;
  value: number;
  expirationDate: string;
  maxUsages: number;
  currentUsages: number;
  active: boolean;
}

File: Shopflow/Shopflow/src/main/java/com/shopflow/shopflow/entity/Coupon.java
package com.shopflow.shopflow.entity;

import com.shopflow.shopflow.enums.CouponType;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "coupons")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Coupon {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 50)
    private String code;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CouponType type;

    /*
     * Si type = PERCENT -> ex: 10 = 10%
     * Si type = FIXED   -> ex: 20 = 20 DT
     */
    @Column(name = "coupon_value", nullable = false)
    private Double value;

    @Column(nullable = false)
    private LocalDateTime expirationDate;

    @Column(nullable = false)
    private Integer maxUsages;

    @Column(nullable = false)
    private Integer currentUsages;

    @Column(nullable = false)
    private Boolean active;
}